### PR TITLE
Tweak: Changed Box Shadow control position in various Core widgets [OBXT-643]

### PR DESCRIPTION
**Pull request by hein-obox**

### CODE REVIEW

**Review by Nevoss — DISMISSED** [review]

File: includes/widgets/image.php [hunk 1]
```diff
@@ -447,11 +458,25 @@ protected function register_controls() {
 			]
 		);
 
+		$this->add_group_control(
+			Group_Control_Box_Shadow::get_type(),
+			[
+				'name' => 'image_box_shadow_hover',
```
Comment (matipojo) [inline includes/widgets/image.php:464]: The old control has a prefix by mistake. So, a new control don't need it.

```suggestion
				'name' => 'shadow_hover',
```

File: includes/widgets/image.php [hunk 2]
```diff
@@ -447,11 +458,25 @@ protected function register_controls() {
...
+				'selector' => '{{WRAPPER}}:hover img',
+			]
+		);
+
 		$this->add_control(
 			'background_hover_transition',
 			[
 				'label' => esc_html__( 'Transition Duration', 'elementor' ),
 				'type' => Controls_Manager::SLIDER,
+				'default' => [
```
Comment (matipojo) [inline includes/widgets/image.php:477]: It will affect existing pages, is it OK?

File: includes/widgets/traits/button-trait.php [hunk 3]
```diff
@@ -341,6 +350,35 @@ protected function register_button_style_controls( $args = [] ) {
 			]
 		);
 
+		$this->add_group_control(
+			Group_Control_Box_Shadow::get_type(),
+			[
+				'name' => 'button_box_shadow_hover',
```
Comment (matipojo) [inline includes/widgets/traits/button-trait.php:356]:
```suggestion
				'name' => 'box_shadow_hover',
```

File: includes/widgets/traits/button-trait.php [hunk 4]
```diff
@@ -341,6 +350,35 @@ protected function register_button_style_controls( $args = [] ) {
...
+				'selector' => '{{WRAPPER}} .elementor-button:hover, {{WRAPPER}} .elementor-button:focus',
+				'condition' => $args['section_condition'],
+			]
+		);
+
+		$this->add_control(
+			'button_transition',
```
Comment (matipojo) [inline includes/widgets/traits/button-trait.php:363]:
```suggestion
			'transition',
```

**Review by matipojo — CHANGES_REQUESTED** [review]

**Comment by rami-elementor** [comment]
Hi @hein-obox 

Can you review the code suggestions and change the branch to `release/3.8.0` ?

We want to promote this PR.

File: includes/widgets/image.php (hunk 2 above)
Comment (rami-elementor) [reply includes/widgets/image.php:477]: Maybe we don't need this default value, especially if it affects existing pages. In any case, we have a default fallback in a CSS variable.

File: includes/widgets/image.php (hunk 1 above)
Comment (hein-obox) [reply includes/widgets/image.php]: @rami-elementor Updated

File: includes/widgets/image.php (hunk 2 above)
Comment (hein-obox) [reply includes/widgets/image.php:477]: @rami-elementor Default control removed

File: includes/widgets/traits/button-trait.php (hunk 3 above)
Comment (hein-obox) [reply includes/widgets/traits/button-trait.php]: Change implemented

@rami-elementor

File: includes/widgets/traits/button-trait.php (hunk 4 above)
Comment (hein-obox) [reply includes/widgets/traits/button-trait.php]: @rami-elementor updated

File: includes/widgets/traits/button-trait.php [hunk 5]
```diff
@@ -340,6 +349,35 @@ protected function register_button_style_controls( $args = [] ) {
...
+					'px' => [
+						'max' => 3,
+						'step' => 0.1,
+					],
+				],
+				'selectors' => [
+					'{{WRAPPER}}' => '--button-transition: {{SIZE}}s',
+				],
+			]
+		);
```
Comment (rami-elementor) [inline includes/widgets/traits/button-trait.php:379]:
```suggestion
		$this->add_control(
			'transition',
			[
				'label' => esc_html__( 'Transition Duration', 'elementor' ),
				'type' => Controls_Manager::SLIDER,
				'default' => [
					'size' => 0.3,
				],
				'size_units' => [ 's' ],
				'range' => [
					's' => [
						'max' => 3,
						'step' => 0.1,
					],
				],
				'selectors' => [
					'{{WRAPPER}}' => '--button-transition: {{SIZE}}{{UNIT}}',
				],
			]
		);
```

Comment (rami-elementor) [reply includes/widgets/traits/button-trait.php:379]: replace `px` with `s` and use `{{SIZE}}{{UNIT}}`.

File: includes/widgets/image.php [hunk 6]
```diff
@@ -447,11 +458,25 @@ protected function register_controls() {
...
+					'size' => 0.3,
+				],
 				'range' => [
 					'px' => [
```
Comment (rami-elementor) [inline includes/widgets/image.php:481]: Same as bellow, replace `px` with `s` and use `{{SIZE}}{{UNIT}}`.

**Pull request closed without merging**